ProductionModePlugin.js

```diff
@@ -191,66 +190,93 @@ ProductionModePlugin.prototype.apply = function(compiler) {
-    //    loaded. So, we accomplish what we need: have the data loaded as soon
-    //    as the chunk is loaded, which means it will be available when each
+    //    included in the entry module of each of these chunks.
+    //    So, we accomplish what we need: have the data loaded as soon as the
+    //    chunk is loaded, which means it will be available when each
     //    individual parent code needs it.
-    //
-    // OBS: `additional-chunk-assets` is used to make globalize-compiled-data
-    // changes right before `optimize-chunk-assets` (used by plugins like
-    // Uglify).
-    //
-    // TODO: Can we do this differently than using regexps?
-    compilation.plugin("additional-chunk-assets", function(chunks) {
+    compilation.plugin("after-optimize-module-ids", function() {
       var globalizeModuleIds = [];
       var globalizeModuleIdsMap = {};
 
-      chunks.forEach(function(chunk) {
+      this.chunks.forEach(function(chunk) {
         chunk.modules.forEach(function(module) {
           var aux;
           var request = module.request;
           if (request && util.isGlobalizeRuntimeModule(request)) {
-            // While request has the full pathname, aux has something like "globalize/dist/globalize-runtime/date".
+            // While request has the full pathname, aux has something like
+            // "globalize/dist/globalize-runtime/date".
             aux = request.split(/[\/\\]/);
             aux = aux.slice(aux.lastIndexOf("globalize")).join("/").replace(/\.js$/, "");
-            globalizeModuleIds.push(module.id);
-            globalizeModuleIdsMap[aux] = module.id;
+
+            // some plugins, like HashedModuleIdsPlugin, may change module ids
+            // into strings.
+            var moduleId = module.id;
+            if (typeof moduleId === "string") {
+              moduleId = JSON.stringify(moduleId);
+            }
+
+            globalizeModuleIds.push(moduleId);
+            globalizeModuleIdsMap[aux] = moduleId;
           }
         });
       });
 
-      chunks.filter(function(chunk) {
-        return /globalize-compiled-data/.test(chunk.name);
-      }).forEach(function(chunk) {
-        var locale = chunk.name.replace("globalize-compiled-data-", "");
-        chunk.files.filter(ModuleFilenameHelpers.matchObject).forEach(function(file) {
-          var isFirst = true;
-          // Match either webpack 1.x or webpack 2.x
-          var source = compilation.assets[file].source().replace(/\n\/\*\*\*\/ (\(?function)\(module, exports(, __webpack_require__)?\) {[\s\S]*?(\n\/\*\*\*\/ })/g, function(garbage1, fnHead, garbage2, fnTail) {
-            var fnContent;
+      // rewrite the modules in the localized chunks:
+      //   - entry module will contain the compiled formatters and parsers
+      //   - non-entry modules will be rewritten to export globalize
+      this.chunks
+        .filter((chunk) => /globalize-compiled-data/.test(chunk.name))
+        .forEach((chunk) => {
```

Comment:
Please, update `package.json` node engine accordingly for such support, i.e., arrow functions, const, etc.